### PR TITLE
Make on demand update duration configurable

### PR DESCRIPTION
**Pull request by dzhulgakov**
What title says. We read it from the main config and apply it to the next iteration of ondemand config loading. It also refactors the background thread loop to do time accounting properly if the update intervals are not divisible

### CODE REVIEW

File: libkineto/src/Config.cpp [hunk 1]
```diff
@@ -42,6 +42,7 @@ constexpr int kDefaultSamplesPerReport(1);
 constexpr int kDefaultMaxEventProfilersPerGpu(1);
 constexpr int kDefaultEventProfilerHearbeatMonitorPeriod(0);
 constexpr seconds kMaxRequestAge(10);
+constexpr seconds kDefaultOnDemandConfigUpdateIntervalSecs(5);
```
Comment (aaronenyeshi) [inline libkineto/src/Config.cpp:45]: Why is this a different default value than the constexpr in configLoader.cpp:39 below? (ie. 5 vs 1)

**Review by aaronenyeshi — APPROVED** [review]
LGTM, may want to add a comment in the update thread loop. I am good with the ability to change config update internal on-demand.

File: libkineto/src/Config.cpp (hunk 1 above)
Comment (dzhulgakov) [reply libkineto/src/Config.cpp:45]: The one below is not even used, let me clean it up

**Comment by dzhulgakov** [comment]
Add some comments too

**Comment by robieta** [comment]
Looks like the merge conflict here is real.

**Comment by dzhulgakov** [comment]
Rebased!

**Review by aaronenyeshi — APPROVED** [review]

**Pull request closed without merging**